config/config.go

```diff
@@ -0,0 +1,350 @@
+				for k := range options {
+					if maxLength < len(k) {
+						maxLength = len(k)
+					}
+				}
+				for k, option := range options {
+					fmt.Printf("  %s.%s %s\n", name, pad(k+":", maxLength), option.Description)
+				}
+			}
+			return false
+		}
+
+		// Find value
+		value = config[name][key]
+	}
+
+	if argv["help"] == true {
+		if option != nil {
+			// Print help for an option
+			printOptionHelp(name, key, *option, value)
+		} else {
+			// Print list of options
+			printHelp()
+		}
+	} else if argv["set"] == true {
+		// Read value from stdin if necessary
+		data := argv["<value>"].(string)
+		if data == "-" {
+			d, err := ioutil.ReadAll(os.Stdin)
+			if err != nil {
+				fmt.Println("Failed to read value from stdin, error: ", err)
+				return false
+			}
+			data = string(d)
+		}
+
+		// Parse value if necessary
+		if option.Parse {
+			err := json.Unmarshal([]byte(data), &value)
+			if err != nil {
+				fmt.Printf("Failed to parse JSON value, error: %s\n", err)
+				return false
+			}
+		} else {
+			value = data
+		}
+
+		// Validate value
+		if option.Validate != nil {
+			if err := option.Validate(value); err != nil {
+				fmt.Println("Invalidate value, error: ", err)
+				return false
+			}
+		}
+
+		// Save option
+		if argv["--dry-run"] == false {
+			config[name][key] = value
+			if err := Save(config); err != nil {
+				fmt.Println("Failed to save configuration file, error: ", err)
+				return false
+			}
+		}
+
+		fmt.Printf("Set '%s.%s' = %s\n", key, name, data)
+	} else if argv["reset"] == true {
+		// Reset a specific option
+		if option != nil {
+			config[name][key] = option.Default
+			fmt.Printf("Reset '%s.%s' to default value\n", name, key)
+		} else {
+			// Reset all options
+			for name, provider := range extpoints.CommandProviders() {
+				for key, option := range provider.ConfigOptions() {
+					config[name][key] = option.Default
+					fmt.Printf("Reset '%s.%s' to default value\n", name, key)
+				}
+			}
+		}
+
+		// Save configuration
+		if err := Save(config); err != nil {
+			fmt.Println("Failed to save configuration file, error: ", err)
+			return false
+		}
+
+	} else {
+		// Select formatter
+		formatter := formatYAML
+		if f, ok := argv["--format"].(string); ok {
+			if f == "json" {
+				formatter = formatJSON
+			} else if f != "yaml" {
+				fmt.Printf("Unsupported output format: %s\n", f)
+				return false
+			}
+		}
+
+		// Open output file
+		out := io.Writer(os.Stdout)
+		if o, ok := argv["--output"].(string); ok {
+			if o != "-" {
+				outFile, err := os.Create(o)
+				if err != nil {
+					fmt.Printf("Failed to create output file '%s' error: %s\n", o, err)
+					return false
+				}
+				defer outFile.Close()
+				out = outFile
+			}
+		}
+
+		// Get all keys, if key was given
+		if option == nil {
+			value = config
+		}
+
+		// Write output
+		if _, err := out.Write(formatter(value)); err != nil {
+			fmt.Println("Error writing result, error: ", err)
+			return false
+		}
+	}
```

Comment:
See [Effective Go](https://golang.org/doc/effective_go.html#switch). This should be

```
switch {
  case argv["help"]:
    ...
  case argv["set"]:
    ...
  case argv["reset"]:
    ...
  default:
    ...
}